pkg/speaker/layer2/arp.go

```diff
@@ -197,6 +204,34 @@ func (a *arpSpeaker) SetBalancer(ip string, nodes []corev1.Node) error {
 	return fmt.Errorf("node %s has no nexthop", nodes[0].Name)
 }
 
+func (a *arpSpeaker) setBalancerFromIPRange(cidr string) error {
+	var err error
+	// convert string to IPNet struct
+	_, ipv4Net, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return err
+	}
+	// convert IPNet struct mask and address to uint32
+	// network is BigEndian
+	mask := binary.BigEndian.Uint32(ipv4Net.Mask)
+	start := binary.BigEndian.Uint32(ipv4Net.IP)
+
+	// find the final address
+	finish := (start & mask) | (mask ^ 0xffffffff)
+
+	// loop through addresses as uint32
+	for i := start; i <= finish; i++ {
+		// convert back to net.IP
+		ip := make(net.IP, 4)
+		binary.BigEndian.PutUint32(ip, i)
+		err = a.setBalancer(ip.String(), []string{ip.String()})
+		if err == nil {
+			return err
+		}
+	}
+	return err
```

Comment:
I failed the test. You should make sure the ip is the node of the cluster before calling setbalancer.